### DISCUSSION TITLE
Is there a possibility of multi-column layout?

### DISCUSSION THREAD

**Discussion by burbridgeconsulting** [GetPublii/Publii#1422] [General]
As a years-long WordPress developer (but note: happy to try something else!), the question always arises: If one wanted to create more of a sophisticated marketing-type site, the questions always arise — is there a way to lay out rows, columns, and fine-tuning of elements, etc. at that level? I am curious. Thanks!

**Comment by example-user30** [comment]
I would love to see a multi-column block in the block editor that is like the one in Gutenberg.  Right now there is not a native way to do this without creating a custom theme or adding custom classes and custom CSS for those classes.

**Reply by burbridgeconsulting** [reply]
I do not know enough yet about what is going on under the hook yet to have ideas about it; but I know that all the modern builders in WordPress use a CSS framework (such as Bootstrap) to actually implement the column layout. The one I use — Beaver Builder — lets you build up rows -> columns (and later, they graduated to sub-columns, which creates a ton of flexibility, but is certainly rather advanced).

But basically it makes it easy to do stuff like:

-------------------------------------------- ROW --------------------------------------------
------------------ column ------------------      ------------------ column ------------------

-------------------------------------------- ROW --------------------------------------------
------------------------------------------- column -------------------------------------------

Etc. (And then of course you need actual content blocks within that.)

**Comment by candidexmedia** [comment]
Short answer: yes! Anything that's possible to do with HTML + CSS + JS on a static site can be done in Publii.

The tricky part is implementing certain desired layouts or design elements manually if there isn't a button or preset for it, especially for users who aren't well-versed in web technologies.

Luckily, the [question has been asked before](https://forum.getpublii.com/topic/front-page-index-how-to-make-a-section-with-3-responsive-column) in the old forum, so I'll share my response to it here as well: https://forum.getpublii.com/topic/front-page-index-how-to-make-a-section-with-3-responsive-column/#post-11731

My suggestion would be to find an example of a column layout that you like on StackOverflow, Codepen, CSS Tricks, MDN, W3, GPT, etc., and porting it to Publii as explained [in my other reply](https://forum.getpublii.com/topic/front-page-index-how-to-make-a-section-with-3-responsive-column/#post-11731). Feel free to ask for help here if you run into any issues.

To save that code snippet as a template that can be reused across your posts, see my [Gist on the Template plugin](https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c#file-2-adding-a-template-plugin-for-reusable-chunks-md):

![](https://camo.githubusercontent.com/5261259f99ff16a2167623e748d78da44ca2a28be94bdda822e7b3ad8b9950d7/68747470733a2f2f692e696d6775722e636f6d2f646f327538754a2e676966)
